site/app/Resources/views/base.html.twig

```diff
@@ -58,18 +58,35 @@
           {% if is_granted("IS_AUTHENTICATED_REMEMBERED") %}
           <li class="dropdown">
             <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">
-              {{ 'layout.logged_in_as'|trans({'%username%': app.user.username}, 'FOSUserBundle') }}
+              {{ 'layout.logged_in_as'|trans({'%username%': app.user.usernameCanonical}, 'FOSUserBundle') }}
```

Comment:
You are right, this was a force of habit of mostly using canonical usernames and templates are not the right place for it.